runtime/common/src/extensions/check_nonce.rs

```diff
@@ -98,11 +98,16 @@ where
 			.into())
 		}
 
+		// Is this an existing account ?
+		let existing_account =
+			account.providers > 0 || account.consumers > 0 || account.sufficients > 0;
```

Comment:
based on the logic on here if all three of these feilds are zero then the account will be reaped
https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/system/src/lib.rs#L1246-L1247.

I could have used `nonce > 0` but that seems to be the side effect of an account existing not the real logic that gets checked.

Still if somebody feels strongly about adding `nonce > 0 ` that can also being added here

Reply:
Is this logic copied from somewhere else? Might be good to include a reference to the source as a comment.

Reply:
This is extracted from the logic mentioned in this comment but will add it in the source also so we know where this came from.